internal/format/format.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package format
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/edgexfoundry/edgex-cli/internal/config"
+	"github.com/edgexfoundry/edgex-cli/internal/service"
+)
+
+type FormatType int
+
+const (
+	Debug FormatType = iota
+	JSON
+	Value
+)
+
+func PrintServiceTable(endpoint string, ft FormatType, jsonValueName string) {
+	PrintServices(config.Configuration.CoreServices, endpoint, ft, jsonValueName)
+	PrintServices(config.Configuration.DeviceServices, endpoint, ft, jsonValueName)
+	PrintServices(config.Configuration.ApplicationServices, endpoint, ft, jsonValueName)
+
+}
+
+func PrintServices(service config.Services, endpoint string, ft FormatType, jsonValueName string) {
+	for serviceName, service := range service {
+
+		var value, jsonValue, url string
+		var err error
+		value, jsonValue, url, err = GetServiceValue(service, endpoint, jsonValueName)
+		if err != nil {
+			if ft == Debug {
+				fmt.Printf("%s: %s: %s\n", serviceName, url, err.Error())
```

Comment:
Why only printing the error when in DEBUG?? Is Debug really Verbose?

Reply:
Yes, as per line 34 of the ADR draft:

```
`-d`, `--debug` | show additional output for debugging purposes (e.g. REST URL, request JSON, …). This command-line arg will replace -v, --verbose and will no longer trigger output of the response JSON (see -j, --json).
```
I could rename the constant as `Verbose` if that reflects the intention better?

Reply:
I like it!